neuralpredictors/layers/modulators/mlp.py

```diff
@@ -26,7 +26,11 @@ def __init__(
             self.modulator_networks.append(nn.Sequential(*feat))
 
     def regularizer(self):
-        return self.linear.weight.abs().mean()
+        reg = 0
+        for mod in self.modulator_networks:
+            # regularize the weights of the last linear layer
+            reg += mod[-2].weight.abs().mean()
```

Comment:
This one seems to make an assumption about the modulator network, right? - that the second last layer is a linear layer, and what is the last one? a nonlinearity? Is there a more general way to find the last linear layer in the network? It might be that the modulator is constructed without a final nonlinearity, no?

Reply:
The way that the `__init__` method is constructed at the moment, the last linear layer will always be the index `-2`. This would break, of course, if the `__init__` was changed in a future PR without taking care of changing the rest of the class. If we want to exclude this possible future bug, how would you implement finding the last linear layer? Something like this?
```
reg = 0
for mod in self.modulator_networks:
    # find the last layer
    for l in range len(mod):
        if isinstance(mod[-l], nn.Linear):
            last_linear_layer = mod[-l]
            break
            
    # regularize the weights of the last linear layer
    reg += last_linear_layer.weight.abs().mean()
```

Reply:
Yes, this looks good - except that the `()` are missing for range. 
I also took a look at the `__init__` method and you are right that it is very specific in how the model is constructed and this way of checking for the last linear layer is probably an overkill. So up to you. I would suggest the second option so that we have a good example for the future and also if someone changed it later then it does the expected thing (i.e. regularize the last linear layer).